pkg/catalog/catalog.go

```diff
@@ -1,18 +1,114 @@
 package catalog
 
 import (
+	"encoding/json"
 	"fmt"
+	// "io/ioutil"
+	// "net/http"
 	"sort"
 	"strings"
+	// "time"
 
 	"github.com/golang/glog"
 	imagev1 "github.com/openshift/api/image/v1"
 	"github.com/openshift/odo/pkg/occlient"
 	"github.com/openshift/odo/pkg/util"
 	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// CheDevfileRegistry is Che devfile registry reposotiry
+const CheDevfileRegistry = "https://che-devfile-registry.openshift.io/"
+
+// DevfileIndexLink is the index.json link of Che devfile registry
+const DevfileIndexLink = CheDevfileRegistry + "/devfiles/index.json"
+
+// ListDevfileComponents lists all the available devfile components
+func ListDevfileComponents() (DevfileComponentTypeList, error) {
+
```

Comment:
suggestion: you can make function here `GetDevfileIndex()`, the idea here is all logic can be broken into smaller tasks and can be reused, it makes code more readable.